### PR TITLE
Spectator implement

### PR DESCRIPTION
**Pull request by ghost**
Make admins can set spectators to other eg helpers want to be turned into spectator, 
or Ops can be forced to be in gmsp to visit adminworld.
and more advantages.

### CODE REVIEW

**Comment by marcocorriero** [comment]
No issues. just a small idea.

**Comment by JeromSar** [comment]
@marcocorriero Please read the contributing guidelines (I don't think you have)

**Comment by marcocorriero** [comment]
#1837 @JeromSar sorry :P

**Comment by ItzLevvie** [comment]
Needs to be formatted to the correct style.

**Comment by Wild1145** [comment]
If the issues are correct this is duplicated functionality.

**Comment by JeromSar** [comment]
Also, lots of formatting.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java [hunk 1]
```diff
@@ -6,16 +6,39 @@
...
 public class Command_spectator extends FreedomCommand
 {
-
-    @Override
-    public boolean run(CommandSender sender, Player playerSender, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
-    {
-        playerSender.setGameMode(GameMode.SPECTATOR);
-        msg("Gamemode set to spectator.");
-        return true;
+    public boolean run(final CommandSender sender, final Player playerSender, final Command cmd, final String commandLabel, final String[] args, final boolean senderIsConsole) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java:13]: Please correct the formatting here

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java [hunk 2]
```diff
@@ -6,16 +6,39 @@
...
+        if (args.length == 0) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java:14]: Formatting needs changing here

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java [hunk 3]
```diff
@@ -6,16 +6,39 @@
...
+            if (this.isConsole()) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java:15]: Formatting also needs changing here

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java [hunk 4]
```diff
@@ -6,16 +6,39 @@
...
+                sender.sendMessage("When used from the console, you must define a target player.");
+                return true;
+            }
+            this.checkRank(Rank.SUPER_ADMIN);
+            playerSender.setGameMode(GameMode.SPECTATOR);
+            this.msg("Gamemode set to Spectator.");
+            return true;
+        }
+        else {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java:24]: Formatting is incorrect here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java [hunk 5]
```diff
@@ -6,16 +6,39 @@
...
+            this.checkRank(Rank.SUPER_ADMIN);
+            if (args[0].equals("-a")) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java:26]: Formatting again here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java [hunk 6]
```diff
@@ -6,16 +6,39 @@
...
+                for (final Player targetPlayer : this.server.getOnlinePlayers()) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java:27]: Formatting here as well please.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java [hunk 7]
```diff
@@ -6,16 +6,39 @@
...
+                    targetPlayer.setGameMode(GameMode.SPECTATOR);
+                }
+                FUtil.adminAction(sender.getName(), "Changing everyone's gamemode to spectator", false);
+                return true;
+            }
+            final Player player = this.getPlayer(args[0]);
+            if (player == null) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java:34]: Formatting here please.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java [hunk 8]
```diff
@@ -6,16 +6,39 @@
...
-        playerSender.setGameMode(GameMode.SPECTATOR);
-        msg("Gamemode set to spectator.");
-        return true;
+    public boolean run(final CommandSender sender, final Player playerSender, final Command cmd, final String commandLabel, final String[] args, final boolean senderIsConsole) {
+        if (args.length == 0) {
+            if (this.isConsole()) {
+                sender.sendMessage("When used from the console, you must define a target player.");
+                return true;
+            }
+            this.checkRank(Rank.SUPER_ADMIN);
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java:19]: Why are you checking if they are a super admin if the command can only be executed by a super admin?

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java [hunk 9]
```diff
@@ -6,16 +6,39 @@
...
+            playerSender.setGameMode(GameMode.SPECTATOR);
+            this.msg("Gamemode set to Spectator.");
+            return true;
+        }
+        else {
+            this.checkRank(Rank.SUPER_ADMIN);
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java:25]: Again, I dont understand why you are checking if they are a super admin when you have to be a super admin to even use the command? Not to mention with this I would not even like the idea of setting everyones rank to Spectator, much less by a super admin...

**Review by Wild1145 — CHANGES_REQUESTED** [review]
Please see the changes requested and make said changes before we can pull this PR.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java (hunk 8 above)
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java:19]: because the command will be protected from ops

**Comment by Commodore64x** [comment]
@marcocorriero Mind reformatting it? When you save in netbeans it should auto-format.

**Comment by ghost** [comment]
@Wild1145 The branch for this PR has been deleted by marco, it is now invalid.

**Comment by Wild1145** [comment]
Due to the original repo having been destroyed, I am closing this PR down as a result.

**Pull request closed without merging**